src/main/java/gel/Storage.java

```diff
@@ -0,0 +1,119 @@
+package gel;
+
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.List;
+import java.util.Scanner;
+
+import gel.task.Deadline;
+import gel.task.Event;
+import gel.task.Task;
+import gel.task.Todo;
+
+/**
+ * Storage deals with loading tasks from the file and saving tasks in the file.
+ */
+public class Storage {
+
+    private String filePath;
+    private String directoryPath;
+
+    /**
+     * Constructor of Storage class.
+     *
+     * @param filePath ensures that tasks are stored in the file path.
+     */
+    public Storage(String filePath) {
+        int fileIndex = filePath.lastIndexOf("/");
+        this.directoryPath = filePath.substring(0, fileIndex);
+        this.filePath = filePath;
+    }
+
+    /**
+     * Checks for file and directory existence.
+     * If they do not exist, create new file/directory.
+     *
+     * @throws IOException when createNewFile() fails when filePath is invalid
+     */
+    public void checkFileExistence() throws IOException {
+        File dataDir = new File(directoryPath);
+        if (!dataDir.exists()) {
+            boolean bool = dataDir.mkdir();
```

Comment:
Should this Boolean variable name be "isDirCreated"? The [java coding standard documentation](https://se-education.org/guides/conventions/java/index.html#naming) states that Boolean variables/method should be named to sound like booleans. I notice similar issues regarding the naming of Boolean variables elsewhere as well, perhaps you can do a quick look through your Boolean variable names! :)